### PR TITLE
Preprocessing initial

### PR DESCRIPTION
**Pull request by FlorianBarkmann**
This PR adds the preprocessing module to CanSig.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,12 +1,13 @@
 [flake8]
 max-line-length = 120
-ignore = E712, W503, E203
+ignore = E712, W503, E203, C901
```
Comment (pawel-czyz) [inline .flake8:3]: Sorry to say that, but turning C901 off is a no-go if we want to maintain reasonable code quality :/

Comment (FlorianBarkmann) [reply .flake8:3]: Hmmm, it complained about DataRecorder._sanitize_adata which really isn't that complex, especially compared to other functions.

**Review by pawel-czyz — CHANGES_REQUESTED** [review]
Thanks Florian, this PR required a lot of effort to do – thanks for taking the responsibility! :slightly_smiling_face: 

However, I think we should refactor the code before merging. The rationale is that in the current shape this code will be difficult to maintain and if we don't refactor it now, we'll probably never find motivation to do it.

As this is a lot of code, maybe we could do a _team_ mini-hackathon aimed at refactoring this? (I'll be available after June 20th).

File: .flake8 (hunk 1 above)
Comment (FlorianBarkmann) [reply .flake8:3]: I guess it counts the number of for loops and if statements

File: requirements.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@
 # setup.cfg
 anndata
 gseapy
+infercnvpy
```
[the remark this answers is not in the document]
Comment (FlorianBarkmann) [reply requirements.txt:5]: For preprocessing we do

File: .flake8 (hunk 1 above)
Comment (pawel-czyz) [reply .flake8:3]: In that case, I'd suggest turning it off just for that method by adding the `# noqa: C901` comment :slightly_smiling_face:

Comment (FlorianBarkmann) [reply .flake8:3]: Sounds good! I didn't know that was an option;)

File: cansig/_preprocessing/_infercnv.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    """
+    Validates if the gene order contains the correct columns and if the genes in the
+    gene order match the gene list.
+    Args:
+        gene_order: DataFram containing the positional gene annotations.
+        gene_list: List of genes present in all AnnDatas.
+    """
+    _EXPECTED_COLUMNS: Set[str] = {_GENE_ANNOTATION.CHROMOSOME, _GENE_ANNOTATION.START, _GENE_ANNOTATION.END}
+
+    if (missing_columns := _EXPECTED_COLUMNS - set(gene_order.columns)) != set():
```
[the remark this answers is not in the document]
Comment (FlorianBarkmann) [reply cansig/_preprocessing/_infercnv.py:90]: If not, would work but I think there is benefit to being explicit in your if statements.

File: cansig/_preprocessing/plotting.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import infercnvpy as cnv  # pytype: disable=import-error
+import scanpy as sc  # pytype: disable=import-error
+from anndata import AnnData  # pytype: disable=import-error
+
+
+def embeddings_counts(adata: AnnData) -> None:
+    """
+    Calculates PCA and UMAP embedding for counts.
+    :param adata: Anndata object
```
[the remark this answers is not in the document]
Comment (FlorianBarkmann) [reply cansig/_preprocessing/plotting.py:9]: Thanks,  I missed that one. For the rest it should be google style.

File: cansig/_preprocessing/main.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    cnv_key: str = "cnv",
+    scoring_dict: Optional[ScoringDict] = None,
+    g2m_genes: Optional[GeneList] = None,
+    s_genes: Optional[GeneList] = None,
+    figure_dir: Optional[Pathlike] = None,
+    copy: bool = False,
+    threshold: float = 0.6,
+    depth: int = 6,
+) -> ad.AnnData:
+    normalize_config = NormalizedConfig()
```
[the remark this answers is not in the document]
Comment (FlorianBarkmann) [reply cansig/_preprocessing/main.py:49]: Good Catch!

File: cansig/_preprocessing/main.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    check_min_malignant_cells,
+    check_min_reference_cells,
+    load_adatas,
+    pop_adatas,
+    NormalizedConfig,
+)
+from cansig.types import Pathlike, ScoringDict, GeneList
+
+
+def preprocessing(
```
[the remark this answers is not in the document]
Comment (FlorianBarkmann) [reply cansig/_preprocessing/main.py:22]: I agree but I feel like that is less user-friendly.

**Pull request merged**